src/Component/Checker/IssuedAtChecker.php

```diff
@@ -14,10 +15,22 @@ final class IssuedAtChecker implements ClaimChecker, HeaderChecker
 {
     private const NAME = 'iat';
 
+    private readonly ClockInterface $clock;
+
     public function __construct(
         private readonly int $allowedTimeDrift = 0,
-        private readonly bool $protectedHeaderOnly = false
+        private readonly bool $protectedHeaderOnly = false,
+        ?ClockInterface $clock = null,
```

Comment:
having this argument as the last one means that you cannot actually remove the nullable type in 4.0 as you cannot make it mandatory without making all other arguments mandatory as well.

Reply:
Hi @stof,

You are right, this will cause issues in the futur.
I created #467 and by setting constructors as private this should resolve that.
By the way, I am not sure the `@deprecated` statement on the `__construct` method makes sense.